src/controllers/events.js

```diff
@@ -29,3 +30,37 @@ exports.getRelatedEvents = async (req, res, next) => {
     .populate('ngo', 'name');
   return res.status(200).json({ success: true, data: relatedEvents });
 };
+
+exports.approveApplicant = async (req, res, next) => {
+  const event = await Event.findById(req.params.id);
+  if (!event) {
+    return next(new ErrorResponse('No event found', 404));
+  }
+  // const eventOwner = event.ngo;
+  // if (eventOwner !== User._id) {
+  //   return next(
+  //     new ErrorResponse(
+  //       'you can only approve or decline applicants of your event',
+  //       400
+  //     )
+  //   );
+  // }
+  const applicant = await Volunteer.findById(req.params.userId);
+  const applicantIndex = event.pendingApplicants.indexOf(applicant._id);
+  if (applicantIndex === -1) {
+    return next(
+      new ErrorResponse('This Volunteer did not apply for this event', 400)
+    );
+  }
+
+  await Event.findOneAndUpdate(
+    { _id: mongoose.Types.ObjectId(req.params.id) },
+    { $push: { approvedApplicants: applicant._id } }
+  );
+  await Event.findOneAndUpdate(
+    { _id: mongoose.Types.ObjectId(req.params.id) },
+    { $pull: { pendingApplicants: applicant._id } }
+  );
+  const updatedEvent = await Event.findById(req.params.id);
+  return res.status(200).json({ success: true, data: updatedEvent });
+};
```

Comment:
Great way of implementing push and pull. I wasn't aware of this and was implementing like below. 
```
const event = await Event.findById(req.params.id)
event.pendingApplicants.push(req.params.userId
```)...etc